zephyr/wrapper.c

```diff
@@ -523,3 +534,21 @@ struct idc **idc_get(void)
 	return NULL;
 }
 #endif
+
+#ifdef CONFIG_LIBRARY
+/* Dummies for unsupported architectures */
+
+/* Platform */
+int platform_init(struct sof *sof)
+{
+	return 0;
+}
+int platform_boot_complete(uint32_t boot_message)
+{
+	return 0;
+}
+
+/* Logging */
+const struct log_source_const_data log_const_sof;
```

Comment:
I think this last commit needs split up a little more per subject so we know why each change is needed.